.travis.yml

```diff
@@ -75,10 +75,8 @@ script:
       sudo PATH=$PATH containerd -log-level debug &> /tmp/containerd-cri.log &
       sudo ctr version ;
       sudo PATH=$PATH GOPATH=$GOPATH critest --runtime-endpoint=/var/run/containerd/containerd.sock --parallel=8 ;
-      exit_code=$? ;
-      test $exit_code -ne 0 && cat /tmp/containerd-cri.log ;
+      test $? -ne 0 && cat /tmp/containerd-cri.log ;
       sudo pkill containerd ;
-      exit $exit_code ;
```

Comment:
What will be the exit code now? What's going on in Travis? :confused:

Reply:
Turns out travis is already collecting exit codes from each command and will exit with failure if *any* of the commands within `script:` have a non-zero return code. Somehow, calling `exit` circumvents the rest of travis processing--you can see from older builds that travis never even got to output its usual "closing" success or fail when the `exit $exit_code ;` call exists in travis.yml! basically the collection of this exit code was unnecessary as travis is already handling that.

Reply:
Documented here: https://docs.travis-ci.com/user/customizing-the-build/#How-does-this-work%3F-(Or%2C-why-you-should-not-use-exit-in-build-steps)

Reply:
Ah! makes sense now, thanks :)